Feature/datatables view filter fix

JVickery-TBS opened the pull request:
Fixes #

### Proposed fixes:

Added ascii encoding to datatables views' jinja2 template which solves errors with special characters.

Decoded and unescaped escape characters in the datatables views ajax callback definition.

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport
- [X] None of the above

Please [X] all the boxes above that apply

wardi commented:
xmlcharref isn't a normal way to encode query parameters. Try `urllib.quote` instead.

JVickery-TBS commented:
@wardi I have replaced the xml char ref replace with urllib.quote. I imported urllib.quote in the environment.py and added it to the jinja2 environment as a global.

The jinja2 template now has utf-8 encoding, as required for the urllib.quote to work. And it still has the ascii encoding for the fanstatic issues which remains.

Finally, as discussed, the decoding of the filters is just a simple unquote and string cast. The filters url parameter for the datatables views pipes all of the filters together and does not separate them, so the single string cast seems fine to use here.

wardi commented on ckanext/datatablesview/templates/datatables/datatables_view.html:
```diff
@@ -1,7 +1,7 @@
 
 {% block page %}
-  {% set filters = request.params.get('filters', '')|e if request.params is defined else request.args.get('filters', '')|e %}
+  {% set filters = url_quote(request.params.get('filters', '').encode('utf-8')).encode('ascii')|e if request.params is defined else url_quote(request.args.get('filters', '').encode('utf-8')).encode('ascii')|e %}
```
instead of adding a new global function to all templates used everwhere, turn this dense bit of mostly-python code into a template helper function in the datatables extension.

JVickery-TBS commented:
@wardi I have added helpers.py into the datatables views ckan extension. And I have written the helper functions in there and made use of them in the jinja template, as well as the ajax callback. The request.params is from the toolkit, in which it will do the try for the params or args (Flask or Pylon), so that is handled already there.

wardi commented on ckanext/datatablesview/templates/datatables/datatables_view.html:
```diff
@@ -1,7 +1,7 @@
 
 {% block page %}
-  {% set filters = request.params.get('filters', '')|e if request.params is defined else request.args.get('filters', '')|e %}
+  {% set filters = h.get_filters(encode=True) %}
```
It looks like you removed the handling of `request.args`. For submitting upstream you'll need both because flask request objects put the parameters in `request.args`

JVickery-TBS replied:
@wardi in my helper functions, I use ckan.plugins.toolkit.request, which has it's params property definition as such:
`@property
    def params(self):
        u''' Special case as Pylons' request.params is used all over the place.
        All new code meant to be run just in Flask (eg views) should always
        use request.args
        '''
        try:
            return super(CKANRequest, self).params
        except AttributeError:
            return self.args`
So that should handle any issues between the Flask and Pylons request objects.

wardi commented:
LGTM.
Please create a PR against github.com/ckan/ckan/ with the same changes and mark it for backporting.
